src/app/sign-in/sign-in.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, AbstractControl, Validators, ValidationErrors } from '@angular/forms';
+
+@Component({
+  selector: 'app-sign-in',
+  templateUrl: './sign-in.component.html',
+  styleUrls: ['./sign-in.component.scss']
+})
+export class SignInComponent implements OnInit {
+  signInForm: FormGroup;
+  private readonly pswrd: RegExp = /^(?=\D*\d)(?=[^a-z]*[a-z])(?=[^A-Z]*[A-Z]).{8,30}$/;
+
+  constructor(private readonly fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.signInForm = this.fb.group({
+      username: this.fb.control('', Validators.required),
+      password: this.fb.control('', [Validators.required, this.psswrdValidator.bind(this)]),
```

Comment:
it not needed to check the password for character/number etc. in sign-in, `required` enough

Reply:
fixed